src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -124,30 +115,55 @@ public Task<bool> NominateProjectAsync(string projectUniqueName, IVsProjectResto
             }
         }
 
-        private void DumpProjectRestoreInfo(PackageSpec packageSpec)
+        private void DumpProjectRestoreInfo(DependencyGraphSpec projectRestoreInfo)
         {
             try
             {
-                var outputPath = packageSpec.RestoreMetadata.OutputPath;
+                var outputPath = projectRestoreInfo.Projects.First().RestoreMetadata.OutputPath;
                 if (!Directory.Exists(outputPath))
                 {
                     Directory.CreateDirectory(outputPath);
                 }
 
-                // Create dg file
-                var dgFile = new DependencyGraphSpec();
-                dgFile.AddRestore(packageSpec.RestoreMetadata.ProjectName);
-                dgFile.AddProject(packageSpec);
-
                 var dgPath = Path.Combine(outputPath, $"{Guid.NewGuid()}.dg");
-                dgFile.Save(dgPath);
+                projectRestoreInfo.Save(dgPath);
             }
             catch (Exception e)
             {
                 _logger.LogError(e.ToString());
             }
         }
 
+        private static DependencyGraphSpec ToDependencyGraphSpec(ProjectNames projectNames, IVsProjectRestoreInfo projectRestoreInfo)
+        {
+            var dgSpec = new DependencyGraphSpec();
+
+            var packageSpec = ToPackageSpec(projectNames, projectRestoreInfo);
+            dgSpec.AddRestore(packageSpec.RestoreMetadata.ProjectName);
+            dgSpec.AddProject(packageSpec);
+
+            if (projectRestoreInfo.ToolReferences != null)
+            {
+                projectRestoreInfo
+                    .ToolReferences
+                    .Cast<IVsReferenceItem>()
+                    .Select(r => ToToolPackageSpec(projectNames, r))
+                    .ToList()
+                    .ForEach(ts =>
+                    {
+                        dgSpec.AddRestore(ts.RestoreMetadata.ProjectName);
```

Comment:
``ts.RestoreMetadata.ProjectUniqueName``

They should be the same right now, but just to be completely correct here, the restore name needs to be the unique name since that is what is used to look it up from the projects section.